### PR TITLE
chore(proto): workaround unsupported cp option on mac

### PR DESCRIPTION
**Pull request by rhybrillou**
## Description

The `cp` command on my machine does not support the `--no-preserve` option. As the comment above the failing line reports issues with file ownership and writability, two additional steps are introduced to ensure the copied files are owned and writable by the calling user.

## Checklist
- [ ] Investigated and inspected CI test results
- [ ] Unit test and regression tests added
- [ ] Evaluated and added CHANGELOG entry if required
- [ ] Determined and documented upgrade steps
- [ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))

If any of these don't apply, please comment below.

### Here I tell how I validated my change

TODO(replace-me)  
Use this space to explain **how you validated** that **your change functions exactly how you expect it**.
Feel free to attach JSON snippets, curl commands, screenshots, etc. Apply a simple benchmark: would the information you
provided convince any reviewer or any external reader that you did enough to validate your change.

It is acceptable to assume trust and keep this section light, e.g. as a bullet-point list.

It is acceptable to skip testing in cases when CI is sufficient, or it's a markdown or code comment change only.  
It is also acceptable to skip testing for changes that are too taxing to test before merging. In such case you are
responsible for the change after it gets merged which includes reverting, fixing, etc. Make sure you validate the change
ASAP after it gets merged or explain in PR when the validation will be performed.  
Explain here why you skipped testing in case you did so.

Have you created automated tests for your change? Explain here which validation activities you did manually and why so.

### Reminder for reviewers

In addition to reviewing code here, reviewers **must** also review testing and request further testing in case the
performed one does not seem sufficient. As a reviewer, you must not approve the change until you understand the
performed testing and you are satisfied with it.

### CODE REVIEW

File: qa-tests-backend/scripts/migrate_protos.sh [hunk 1]
```diff
@@ -9,4 +9,6 @@ cp -r ../proto/* ${JAVA_PATH}
 # Migrate v1 API protos from the Scanner repo
 SCANNER_DIR=$(go list -f '{{.Dir}}' -m github.com/stackrox/scanner)
 # files from gomod cache have no write permission causing problems for gradle
-cp -r --no-preserve=mode ${SCANNER_DIR}/proto/* ${JAVA_PATH}
+cp -r ${SCANNER_DIR}/proto/* ${JAVA_PATH}
+chown -R $(whoami) ${JAVA_PATH}
```
Comment (janisz) [inline qa-tests-backend/scripts/migrate_protos.sh:13]: do we need this line?

File: qa-tests-backend/scripts/migrate_protos.sh [hunk 2]
```diff
@@ -9,4 +9,6 @@ cp -r ../proto/* ${JAVA_PATH}
 # Migrate v1 API protos from the Scanner repo
 SCANNER_DIR=$(go list -f '{{.Dir}}' -m github.com/stackrox/scanner)
 # files from gomod cache have no write permission causing problems for gradle
-cp -r --no-preserve=mode ${SCANNER_DIR}/proto/* ${JAVA_PATH}
```
Comment (janisz) [inline qa-tests-backend/scripts/migrate_protos.sh:12]: So looks like mac has old version https://lists.gnu.org/archive/html/bug-coreutils/2003-02/msg00051.html

File: qa-tests-backend/scripts/migrate_protos.sh (hunk 1 above)
Comment (rhybrillou) [reply qa-tests-backend/scripts/migrate_protos.sh:13]: I'm not sure. I think `cp` should create the files with the caller as owner. The ownership change is an additional safety measure.

File: qa-tests-backend/scripts/migrate_protos.sh (hunk 2 above)
Comment (rhybrillou) [reply qa-tests-backend/scripts/migrate_protos.sh:12]: Thanks for checking.

**Review by janisz — APPROVED** [review]

**Pull request merged**